flocker/control/_persistence.py

```diff
@@ -163,15 +245,51 @@ def __init__(self, reactor, path):
     def startService(self):
         if not self._path.exists():
             self._path.makedirs()
-        self._config_path = self._path.child(b"current_configuration.v1.json")
+        self.load_configuration()
+        MultiService.startService(self)
+        _LOG_STARTUP(configuration=self.get()).write(self.logger)
+
+    def load_configuration(self):
+        """
+        Load the persisted configuration, upgrading the configuration format
+        if an older version is detected.
+        """
+        self._config_path = self._path.child(b"current_configuration.json")
```

Comment:
This seems to be a constant.  It might be a good thing to make a property or return from a helper function or to compute at `__init__` time when `_path` is learned.